custom_metrics/almanac.js

```diff
@@ -424,6 +424,33 @@ return JSON.stringify({
 
     return inputNodes;
   })(),
+  'link_protocols_used': (() => {
+    const protocols_and_count = {};
+    const anchors = document.querySelectorAll('a[href]');
+    for (const anchor of anchors) {
+      try {
+        const url = new URL(anchor.href);
```

Comment:
Are you only interested in absolute URLs? Or for example if the a[href] value is `/foo` do you want to consider that as having a protocol of `https` if the base page is also HTTPS?
```suggestion
        const url = new URL(anchor.href, location.href);
```

Reply:
The `href` property on anchor elements already does this for you. You always get the fully qualified url

Reply:
WOW that's cool :)